### ISSUE TITLE
Update dataclay-proxy to use last version of JARs and dependencies

### ISSUE THREAD

**Issue by alexbarcelo** [mF2C/dataClay#17]
There has been a public beta release of dataClay and there have been some changes which may affect the dataclay-proxy image.

Trying to be comprehensive, things that have to be changed:
- [ ] Use new dataClay JAR. Available in the docker images, just like before, but...
- [ ] Include all JAR dependencies. Instead of a single file "fat-jar", now there is a lighter JAR for dataClay and all the dependencies. The tool (in this repo at /tool/dClayTool.sh) has been modified and should address correctly the retrieve of dependencies from the image.
- [ ] Update the classpath to use the dependencies, as well as the dataclay JAR.
- [x] Update the environment variable `DATASERVICE_JAVA_PORT_TCP` (I may have missed some ones).
- [ ] Update path on the docker-compose (from `/usr/src/app` to `/usr/src/dataclay`)

(Last one was just reminded to me right now, I will fix it soon).

**Comment by alexbarcelo** [comment]
The demos folder contains example on the classpath, it will become something like:

```
LIBDIR=../../tool/lib
CLASSPATH=stubs:bin:$LIBDIR/dataclayclient.jar:$LIBDIR/dependencies/*:$CLASSPATH
```

Note both the dataclayclient.jar _plus_ the full dependencies folder (retrieved by the tool automatically).

**Comment by cjdcordeiro** [comment]
@alexbarcelo did you mean to assign this to me?

**Comment by alexbarcelo** [comment]
@cjdcordeiro You or someone on SixSq. I assume that the dataclay-proxy image is maintained by SixSq, as there is some glue magic with clojure that we have no expertise on, and sixsq/dataclay is a fork that has been more recently updated than it one.

**Comment by cjdcordeiro** [comment]
@alexbarcelo we probably need to discuss this during a call. From a business stand point, there's no ground for SixSq to be maintaining something entirely built on top of DC. After all this is the "dataclay"-proxy. We can help of course, because as you say this first version was built in Clojure and that's probably not your preferred implementation. Nevertheless, the only custom content we've done is in https://github.com/mF2C/dataClay/tree/master/dataclay-proxy/src/com/sixsq/dataclay and that's pretty generic and completely unrelated to CIMI. 

I'm happy to help with this, but we cannot maintain it.

If you want we can follow up on this during a skype call or something.

**Comment by alexbarcelo** [comment]
If you can tell us where the CLASSPATH should be changed for the dataclay-proxy build image procedure, then we can maintain that. Also if there is some impact on the tool run, because there is a pre-step to get libraries, which is connected to the classpath but prior to image build, I assume.

The README.md points to release.sh script, is that the general approach, correct?

**Comment by cjdcordeiro** [comment]
the release.sh is the script that does all. 

It uses your DC deployment + registration scripts...then builds the rest API uberjar and builds the docker image. 

It's just an automated script. There's nothing special about the build itself.

**Comment by alexbarcelo** [comment]
@cjdcordeiro I have a bunch of lines like the following:

```
Could not transfer artifact com.sixsq.slipstream:utils:jar:3.53-SNAPSHOT from/to enterprise-snapshots (https://nexus.sixsq.com/content/repositories/snapshots-enterprise-rhel7/): Not authorized , ReasonPhrase:Unauthorized.
```

I was trying to execute the `release.sh` with some modifications. No idea what is happening, no idea what does `lein` do.

Could you enlighten us[me] on that?

Also: it is not finding `dataclay:client:jar:0.0.1` and... where does it come from? release script should do its magic? Or am I lacking some package? I am not in my regular development environment right now, but not sure if it is related to my development environment or your development environment.

**Comment by cjdcordeiro** [comment]
@alexbarcelo hey, @loomis just looked into it. The best is to remove "-SNAPSHOT" from the utils dependency. That package is in the public repository, you should have no problem downloading it. The thing is that SNAPSHOTs are removed with every release so that's why you can't get it.

The only thing which is in a private repository is the dataclay:client:jar:0.0.1, cause we didn't want to put your jar public. That's why we prefer you maintain this. Now you can just point it to your internal repository. 

Lein builds the uberjar.

So in summary, the only changes you need are: 
 - point to a stable utils release instead of SNAPSHOT
 - point to your repository where to find the dataclay jar instead of SixSq's private one

then the release.sh should do its job.
